### ISSUE TITLE
[Bug]: `Select ... limit 0, 18446744073709551615`  in MO throws an error `only int64 support in limit/offset clause` while the same sql executes successfully in both mysql `5.7` and `8.0`

### ISSUE THREAD

**Issue by aronchanisme** [matrixorigin/matrixone#16624]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.2-dev

### Commit ID

d36f88943

### Other Environment Information

```Markdown
- Hardware parameters: 8c 16g
- OS type: All Linux and Mac Distrubutions
- Others: all
```

### Actual Behavior

1. In MO
```bash
github@test0-10-222-4-0:/data/mo/1.2-dev/matrixone$ mo_ctl connect
2024-06-04 14:18:55.439 UTC+0800    [INFO]    Checking connectivity
2024-06-04 14:18:55.529 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 4662
Server version: 8.0.30-MatrixOne-v3688943 MatrixOne

Copyright (c) 2000, 2024, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> create database test;
Query OK, 1 row affected (0.02 sec)

mysql> use test;
Database changed
mysql> show tables;
Empty set (0.00 sec)

mysql> create table t1(id int);
Query OK, 0 rows affected (0.03 sec)

mysql> select * from t1 limit 0, 18446744073709551615;
ERROR 1149 (HY000): SQL syntax error: only int64 support in limit/offset clause
mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| d36f88943     |
+---------------+
1 row in set (0.00 sec)
```

2. In MySQL `5.7.42`
```bash
root@mo-sz-0:~# mysql57 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 2
Server version: 5.7.42 MySQL Community Server (GPL)

Copyright (c) 2000, 2024, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> create database test;
Query OK, 1 row affected (0.00 sec)

mysql> use test;
Database changed
mysql> create table t1(id int);
Query OK, 0 rows affected (0.01 sec)

mysql> select * from t1 limit 0, 18446744073709551615;
Empty set (0.00 sec)

mysql> select version();
+-----------+
| version() |
+-----------+
| 5.7.42    |
+-----------+
1 row in set (0.00 sec)
```

3. In MySQL `8.0.31`
```bash
root@mo-sz-0:~# mysql80 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 9
Server version: 8.0.31 MySQL Community Server - GPL

Copyright (c) 2000, 2024, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> use test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> drop table t1;
Query OK, 0 rows affected (0.02 sec)

mysql> create table t1(id int);
Query OK, 0 rows affected (0.03 sec)

mysql> select * from t1 limit 0, 18446744073709551615;
Empty set (0.00 sec)

mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.31    |
+-----------+
1 row in set (0.00 sec)
```

### Expected Behavior

MO behaves the same as what MySQL does.

### Steps to Reproduce

```Markdown
create database if not exists test;
use test;
drop table if exists t1;
create table t1(id int);
select * from t1 limit 0, 18446744073709551615;
```

### Additional information

As per https://dev.mysql.com/doc/refman/8.4/en/select.html,  `limit` can take up to 2 parameters (Offset, Limit) , and the `limit` value can be set to a very large one in order to retrieve all rows from a table(in case you don't know the number of rows in that table).
> To retrieve all rows from a certain offset up to the end of the result set, you can use some large number for the second parameter. This statement retrieves all rows from the 96th row to the last:

> SELECT * FROM tbl LIMIT 95,18446744073709551615;

**Comment by daviszhen** [comment]
数值越界

**Comment by aronchanisme** [comment]
`2^64 = 18446744073709551616` which is exactly `1` larger than the given limit number `18446744073709551615`.
In MySQL 8.0.31, it seems like only limit number less than `18446744073709551615` is allowed, otherwise mysql will thows an `syntax error` expection.

```bash
mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.31    |
+-----------+
1 row in set (0.00 sec)

mysql> select * from t1 limit 0, 18446744073709551616;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '18446744073709551616' at line 1
mysql> select * from t1 limit 0, 18446744073709551615;
Empty set (0.00 sec)

mysql> select * from t1 limit 0, 18446744073709551617;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '18446744073709551617' at line 1
mysql> select * from t1 limit 0, 18446744073709551614;
Empty set (0.00 sec)

mysql> select * from t1 limit 0, 18446744073709551615;
Empty set (0.00 sec)

mysql> select * from t1 limit 0, 18446744073709551618;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '18446744073709551618' at line 1
```

Corresponding mo source code: https://github.com/matrixorigin/matrixone/blob/b076d9c50305e2bdbafcf9669633a02f0831e70f/pkg/sql/plan/limit_binder.go#L88

**Comment by daviszhen** [comment]
pr中

**Comment by daviszhen** [comment]
pr中

**Comment by daviszhen** [comment]
pr中

**Comment by aronchanisme** [comment]
Fixed in `main`(`1790d7e63`) and `1.2-dev`(`4d3b371e4`). Bvt [cases](https://github.com/matrixorigin/matrixone/blob/main/test/distributed/cases/dml/select/limit.sql) are also added. Closing it as completed.

1. `main`
```bash
github@test0-10-222-4-0:/data/mo/main/matrixone$ mo_ctl connect
2024-06-09 16:19:51.092 UTC+0800    [INFO]    Checking connectivity
2024-06-09 16:19:51.176 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 3209
Server version: 8.0.30-MatrixOne-v1790763 MatrixOne

Copyright (c) 2000, 2024, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> create database if not exists test;
Query OK, 1 row affected (0.03 sec)

mysql> use test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> drop table if exists t1;
Query OK, 0 rows affected (0.06 sec)

mysql> create table t1(id int);

Query OK, 0 rows affected (0.02 sec)

mysql> select * from t1 limit 0, 18446744073709551615;
Empty set (0.00 sec)

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 1790d7e63     |
+---------------+
1 row in set (0.00 sec)

mysql> system mo_ctl get_branch
2024-06-09 16:20:19.885 UTC+0800    [INFO]    Try get mo branch
2024-06-09 16:20:19.939 UTC+0800    [INFO]    Get branch succeeded, current branch: main
mysql> system mo_ctl get_cid
2024-06-09 16:20:24.109 UTC+0800    [INFO]    Try get mo commit id
commit 1790d7e63bfaf1e28b942e9fbc2684c1ae6a8637
Author: gouhongshen <gouhongshen@hotmail.com>
Date:   Sat Jun 8 15:57:35 2024 +0800

    refactor the block reader filter to support more expressions (#16756)
    
    1.  `>, >=, <, <=`
    2. `between and, prefix in, prefix between, prefix eq`
    3. `in, eq`
    
    Approved by: @XuPeng-SH, @aressu1985
2024-06-09 16:20:24.161 UTC+0800    [INFO]    Get commit id succeeded
```

2. `1.2-dev`
```bash
github@test0-10-222-4-0:/data/mo/1.2-dev/matrixone$ mo_ctl connect
2024-06-09 16:25:30.801 UTC+0800    [INFO]    Checking connectivity
2024-06-09 16:25:30.896 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 6302
Server version: 8.0.30-MatrixOne-v433714 MatrixOne

Copyright (c) 2000, 2024, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> create database if not exists test;
Query OK, 1 row affected (0.03 sec)

mysql> use test;

Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> drop table if exists t1;
Query OK, 0 rows affected (0.04 sec)

mysql> create table t1(id int);
Query OK, 0 rows affected (0.02 sec)

mysql> select * from t1 limit 0, 18446744073709551615;
Empty set (0.00 sec)

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 4d3b371e4     |
+---------------+
1 row in set (0.01 sec)

mysql> system mo_ctl get_branch
2024-06-09 16:25:46.208 UTC+0800    [INFO]    Try get mo branch
2024-06-09 16:25:46.265 UTC+0800    [INFO]    Get branch succeeded, current branch: 1.2-dev
mysql> system mo_ctl get_cid
2024-06-09 16:25:48.870 UTC+0800    [INFO]    Try get mo commit id
commit 4d3b371e492dc8f3caef12a2f0ac96cfdc8c3f7a
Author: reusee <reusee@gmail.com>
Date:   Sun Jun 9 15:29:30 2024 +0800

    fileservice: fix fd leak in LocalFS.read (#16749)
    
    fix fd leak in LocalFS.read
    
    Approved by: @fengttt, @sukki37
2024-06-09 16:25:48.936 UTC+0800    [INFO]    Get commit id succeeded
```
